test/language/class/subclass/class-definition-superclass-generator.js

```diff
@@ -0,0 +1,21 @@
+// Copyright (C) 2014 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+es6id: 14.5.14_S6.g.i
+description: >
+    Runtime Semantics: ClassDefinitionEvaluation
+
+    If superclass has a [[FunctionKind]] internal slot whose value is "generator", throw a TypeError exception.
+---*/
+function * G() {}
+
+class A extends G {
```

Comment:
The `TypeError` is already thrown during 14.5.14 ClassDefinitionEvaluation, that means the complete class declaration needs to be moved into the `assert.throws` function. (And `new A();` can be removed.)